File: frontend/src/api/analysis.ts
import client from './client'
import type { APIResponse } from '@/types/common'

export interface SummaryData {
  total_records: number
  total_batches: number
  by_type: Array<{
    data_type: string
    record_count: number
    anomaly_count: number
    anomaly_rate: number
    avg_quality_score: number
  }>
}

export interface TrendData {
  metric: string
  data_points: Array<{
    period: string
    value: number
    data_type: string
  }>
}

export interface QualityData {
  total_records: number
  missing_rate: number
  anomaly_rate: number
  avg_quality_score: number
  by_department: Array<{
    department: string
    total_records: number
    anomaly_rate: number
    avg_quality_score: number
  }>
}

export interface DashboardData {
  summary: SummaryData
  trend: TrendData
  quality: QualityData
}

export const analysisApi = {
  dashboard() {
    return client.get<APIResponse<DashboardData>>('/analysis/dashboard')
  },

  summary() {
    return client.get<APIResponse<SummaryData>>('/analysis/summary')
  },

  trend(metric = 'record_count', dataType = '', granularity = 'month') {
    return client.get<APIResponse<TrendData>>('/analysis/trend', {
      params: { metric, data_type: dataType, granularity },
    })
  },

  quality() {
    return client.get<APIResponse<QualityData>>('/analysis/quality')
  },

  exportData(params: Record<string, any>) {
    return client.post<APIResponse<{ filename: string; download_url: string }>>('/analysis/export', params)
  },
}

File: app/schemas/analysis.py
from datetime import date, datetime
from typing import Any, Optional
from pydantic import BaseModel


class SummaryItem(BaseModel):
    """按 data_type 分组的汇总统计项。"""

    data_type: str
    record_count: int
    date_range_start: Optional[str] = None
    date_range_end: Optional[str] = None
    anomaly_count: int
    anomaly_rate: float
    avg_quality_score: float


class SummaryResponse(BaseModel):
    """汇总统计响应。"""

    total_records: int
    total_batches: int
    by_type: list[SummaryItem]


class TrendPoint(BaseModel):
    """趋势数据点。"""

    period: str  # 如 "2025-11"
    value: float
    data_type: str


class TrendResponse(BaseModel):
    """趋势分析响应。"""

    metric: str  # 如 "record_count" / "anomaly_rate"
    data_points: list[TrendPoint]


class QualityReportResponse(BaseModel):
    """数据质量报告。"""

    total_records: int
    missing_rate: float  # 缺失率
    anomaly_rate: float  # 异常率
    avg_quality_score: float
    by_department: list[dict[str, Any]]


class ExportRequest(BaseModel):
    """导出请求。"""

    data_types: Optional[list[str]] = None  # 如 ["attendance", "sales"]
    date_start: Optional[str] = None
    date_end: Optional[str] = None
    include_anomalies: bool = True
